server/src-lib/Hasura/RQL/Types/Common.hs

```diff
@@ -292,12 +310,8 @@ unsafeNonNegativeInt = NonNegativeInt
 instance FromJSON NonNegativeInt where
   parseJSON = withScientific "NonNegativeInt" $ \t -> do
     case t >= 0 of
-      True  -> NonNegativeInt <$> maybeInt (toBoundedInteger t)
+      True  -> maybe (fail "integer passed is out of bounds") (pure . NonNegativeInt) $ toBoundedInteger t
```

Comment:
```suggestion
      True  -> NonNegativeInt <$> onNothing (toBoundedInteger t) $ fail "integer passed is out of bounds"
```
(not sure about the binding strength of `<$>` vs `$`)

Reply:
Yeah, I used `maybe` on purpose here, since the positive case is not just a `pure`; I find it more readable in this case?

Reply:
Okay, I don't feel strongly about this.